torch/csrc/jit/tracer.cpp

```diff
@@ -48,7 +48,13 @@ void ArgumentStash::stashIntListElem(const std::string& arg_name, size_t size, s
   JIT_ASSERT(size == list_trace.size());
   JIT_ASSERT(idx < list_trace.size());
   JIT_ASSERT(list_trace[idx] == nullptr);
-  list_trace[idx] = getValueTrace(tracing_state, var);
+
+  Value* ten = getValueTrace(tracing_state, var);
+  auto& g = *ten->owningGraph();
+  auto prim = g.createTensorToNum(jit::IntType::get(), ten)
```

Comment:
It can't be a number because it is being converted from a Variable. The only thing we can trace right now is Variable, and all IR nodes attached to Tensors will have Tensor types. This is the other half of why sizeOf does NumToTensor. We can consider creating a "recording primitive" python type that we return instead in these situations, but that is a lot more work and still would require the Argument stash to stick around.